feat: add C implementation for `stats/base/dists/signrank/quantile`

jalajk3004 opened the pull request:
### **Resolves** 
https://github.com/stdlib-js/stdlib/issues/3885

## Description

> What is the purpose of this pull request?

This pull request:

- Adds a C implementation of the Quantile for a signrank distribution.
- Includes benchmarks for the C implementation.
- Provides examples demonstrating the usage of the Quantile for a Signrank distribution in C.


## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolving https://github.com/stdlib-js/stdlib/pull/4154

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

stdlib-bot committed:
docs: update related packages sections

PR-URL: https://github.com/stdlib-js/stdlib/pull/4728
Reviewed-by: Athan Reines <kgryte@gmail.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

stdlib-bot committed:
docs: update namespace TypeScript declaration comments

PR-URL: https://github.com/stdlib-js/stdlib/pull/4729
Reviewed-by: Athan Reines <kgryte@gmail.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

jalajk3004 committed:
adding c implementaion in readme.md

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

jalajk3004 committed:
Added a benchmark file

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: failed
---

jalajk3004 committed:
feat:add c implementation example file

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: failed
---

aayush0325 committed:
fix: add `math/base/speical/abs` in manifest.json of `blas/ext/base/dapxsumkbn`

PR-URL: https://github.com/stdlib-js/stdlib/pull/4732

Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

aayush0325 committed:
docs: update examples of `stats/base/dmin`

PR-URL: https://github.com/stdlib-js/stdlib/pull/4713

Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

jalajk3004 committed:
feat:add the C src files

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

jalajk3004 committed:
changes in package.json

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

jalajk3004 committed:
feat:added test files

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: failed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: failed
---

jalajk3004 committed:
feat:added test files

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: failed
---

jalajk3004 committed:
feat:added test files

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: failed
---

jalajk3004 committed:
feat:added test files

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: failed
---

jalajk3004 committed:
feat:added test files

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: failed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: failed
---

aayush0325 committed:
feat: add C ndarray interface and refactor implementation for `stats/base/dmeankbn2`

PR-URL: https://github.com/stdlib-js/stdlib/pull/4715

Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

headlessNode committed:
refactor: update `blas/ext/base/dapx` to follow current project conventions

PR-URL: https://github.com/stdlib-js/stdlib/pull/4737

Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

stdlib-bot committed:
docs: update related packages sections

PR-URL: https://github.com/stdlib-js/stdlib/pull/4739

Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

0PrashantYadav0 committed:
feat: add C implementation for `stats/base/dists/geometric/skewness`

PR-URL: https://github.com/stdlib-js/stdlib/pull/4093
Closes: https://github.com/stdlib-js/stdlib/issues/3641

Co-authored-by: Philipp Burckhardt <pburckhardt@outlook.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>
Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

stdlib-bot committed:
docs: update REPL namespace documentation

PR-URL: https://github.com/stdlib-js/stdlib/pull/4741
Reviewed-by: Athan Reines <kgryte@gmail.com>
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>

stdlib-bot committed:
docs: update namespace table of contents

PR-URL: https://github.com/stdlib-js/stdlib/pull/4742
Reviewed-by: Athan Reines <kgryte@gmail.com>

stdlib-bot committed:
docs: update namespace TypeScript declarations

PR-URL: https://github.com/stdlib-js/stdlib/pull/4740
Reviewed-by: Athan Reines <kgryte@gmail.com>

anandkaranubc committed:
bench: refactor random number generation in `stats/base/dists/arcsine`

PR-URL: https://github.com/stdlib-js/stdlib/pull/4744
Reviewed-by: Athan Reines <kgryte@gmail.com>

Neerajpathak07 committed:
feat: add C implementation for `number/float32/base/signbit`

PR-URL: https://github.com/stdlib-js/stdlib/pull/3399
Co-authored-by: Athan Reines <kgryte@gmail.com>
Reviewed-by: Athan Reines <kgryte@gmail.com> 
Reviewed-by: Philipp Burckhardt <pburckhardt@outlook.com>
Co-authored-by: stdlib-bot <82920195+stdlib-bot@users.noreply.github.com>

headlessNode committed:
feat: add C `ndarray` API and refactor `blas/ext/base/sapxsumors`

PR-URL: https://github.com/stdlib-js/stdlib/pull/4746
Co-authored-by: Athan Reines <kgryte@gmail.com>
Reviewed-by: Athan Reines <kgryte@gmail.com>

jalajk3004 committed:
refactoring C implementation

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: passed
  - task: run_c_examples
    status: failed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: passed
  - task: run_c_examples
    status: failed
---

jalajk3004 committed:
fix: edit the header file

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: passed
  - task: run_c_examples
    status: failed
---

jalajk3004 commented:
@anandkaranubc I hope this message finds you well. I kindly request your review of my PR [#4734](https://github.com/stdlib-js/stdlib/pull/4734). If you find any issues or areas for improvement, please let me know. I am more than happy to make the necessary changes. I really want my PR to get merged

Thank you for your time and guidance!

jalajk3004 committed:
fix:refactor c implementation

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
---

anandkaranubc commented:
Hi @jalajk3004, it seems your PR includes unrelated changes from the `stdlib` repo, likely introduced when you updated your branch with new changes from `develop`. To fix this, rebase your branch onto the latest `develop` to exclude unrelated changes and confirm that only relevant files are included in the PR. After rebasing, review the changes and push (might have to force-push) the updated branch to clean up the PR.

Let me know when you are done with this and I will revisit the PR again :)

jalajk3004 committed:
fix

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

jalajk3004 commented:
@anandkaranubc I have followed your instruction, Rebase my branch `quantile` onto the latest `develop` branch to exclude unrelated changes and confirm that only relevant files are included in the PR. and did the force-push.
But still there are errors. Kindly look into it and guide me

yaswanthkosuru commented:
@jalajk3004  Before Making PR please ensure your files changes .
Here it was showing 150 files changed it seems  like lot of changes which  leads doesnt merge these P.R. You can Close These  and open new one with consisting of only files under these folder Or you can Modify these one to only include file changes to ~20. Correct me if i am wrong

anandkaranubc commented:
Or you can update your current branch by using the GitHub UI only. Just go to your forked repo, select your branch, sync it and then pull the changes locally.
